raiden-ts/src/raiden.ts

```diff
@@ -782,27 +789,66 @@ export class Raiden {
               }),
             ),
             // request transfer with returned/validated metadata at 'merge' subscription time
-            of(
-              transfer(
-                {
-                  tokenNetwork,
-                  target,
-                  amount: amount as UInt<32>,
-                  metadata,
-                  paymentId,
-                  secret,
-                },
-                { secrethash },
-              ),
-            ).pipe(
-              tap(request => this.store.dispatch(request)),
-              ignoreElements(),
-            ),
+            defer(() => {
+              this.store.dispatch(
+                transfer(
+                  {
+                    tokenNetwork,
+                    target,
+                    amount: value,
+                    metadata,
+                    paymentId,
+                    secret,
+                  },
+                  { secrethash },
+                ),
+              );
+              return EMPTY;
+            }),
           ),
         ),
       )
       .toPromise();
   }
+
+  /**
+   * Request a path from PFS
+   *
+   * If a direct route is possible, it'll be returned. Else if PFS is set up, a request will be
+   * performed and the cleaned/validated result metadata containing the 'routes' will be resolved.
+   * Else, if no route can be found, promise is rejected with respective error.
+   *
+   * @param token - Token address on currently configured token network registry
+   * @param target - Target address (must be getAvailability before)
+   * @param amount - Minimum capacity required on routes
+   * @returns A promise to transfer's secrethash (unique id) when it's accepted
```

Comment:
Does not match with actual return type

Reply:
Ops, forgot this docstring. Will fix, as well as the failing test